state_test.go

```diff
@@ -0,0 +1,171 @@
+package mls
+
+import (
+	"fmt"
+	"testing"
+)
+
+var (
+	groupId   = []byte{0x01, 0x02, 0x03, 0x04}
+	userId    = []byte{0x04, 0x05, 0x06, 0x07}
+	suite     = P256_SHA256_AES128GCM
+	scheme    = Ed25519
+	groupSize = 5
+	setupDone = false
+
+	identityPrivs  []SignaturePrivateKey
+	credentials    []Credential
+	initPrivs      []HPKEPrivateKey
+	clientInitKeys = make([]ClientInitKey, groupSize)
+	states         []State
+
+	testMessage = unhex("1112131415")
+)
+
+func setup(t *testing.T) {
+	if setupDone {
+		return
+	}
+
+	for i := 0; i < groupSize; i++ {
+		// cred gen
+		sigPriv, _ := scheme.Generate()
+		basicCredential = &BasicCredential{
+			Identity:           userId,
+			SignatureScheme:    scheme,
+			SignaturePublicKey: sigPriv.PublicKey,
+		}
+		credentialBasic = Credential{Basic: basicCredential, privateKey: &sigPriv}
```

Comment:
How does this even build?  Seems like `basicCredential` and `credentialBasic` are undefined.

I would collapse this: 
```
cred := &BasicCredential{Basic: &BasicCredential{...}, privateKey: sigPriv}
```

Or just make a `newBasicCredential` method, analogous to `Credential::basic()` in mlspp.

Reply:
because its coming from messages_test.go . let me make a local one here